app/src/main/java/io/github/pietrocaselani/moviestraker/ui/searchmovies/SearchMoviesViewModel.kt

```diff
@@ -0,0 +1,192 @@
+package io.github.pietrocaselani.moviestraker.ui.searchmovies
+
+import android.databinding.BaseObservable
+import android.databinding.ObservableField
+import android.databinding.ObservableInt
+import io.github.pietrocaselani.moviestraker.R
+import io.github.pietrocaselani.moviestraker.entities.MovieEntity
+import io.github.pietrocaselani.moviestraker.helpers.mapMovieToDetails
+import io.github.pietrocaselani.moviestraker.helpers.mapToMovieEntity
+import io.github.pietrocaselani.moviestraker.helpers.mapToMovieListViewModel
+import io.github.pietrocaselani.moviestraker.tmdb.entities.GenreResponse
+import io.github.pietrocaselani.moviestraker.tmdb.entities.ImageConfigurationResponse
+import io.github.pietrocaselani.moviestraker.ui.common.MovieListViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+
+/**
+ * Created by pc on 25/06/17.
+ */
+class SearchMoviesViewModel(private val interactor: SearchMoviesInteractorInput) : BaseObservable() {
+
+	//region Data Binding Properties
+	val movies = ObservableField<MutableList<MovieListViewModel>>(mutableListOf())
+	val message = ObservableField<String>("")
+	val resourceMessage = ObservableInt(0)
+	val moviesVisibility = ObservableField<Boolean>(true)
+	val messageVisibility = ObservableField<Boolean>(true)
+	val selectedMovie = ObservableField<MovieEntity>()
+	//endregion
+
+	//region Properties
+	private val disposables = CompositeDisposable()
+
+	private var imageConfiguration: ImageConfigurationResponse? = null
+	private val movieEntities = mutableListOf<MovieEntity>()
+	private var genres = listOf<GenreResponse>()
+	private val loadedPages = mutableListOf<Int>()
+	private var currentPage = 1
+	private var totalPages = Int.MAX_VALUE
+	private var loading = false
+	private var query = ""
+	//endregion
+
+	//region Lifecycle
+	fun onStart() {
+		if (movieEntities.size == 0) {
+			showMessage(R.string.type_movie_name)
+		}
+
+		fetchGenres()
+	}
+
+	fun onStop() {
+		disposables.clear()
+	}
+	//endregion
+
+	//region Data Binding Methods
+	fun search(query: String) {
+		this.query = query
+		this.currentPage = 1
+		loadedPages.clear()
+		loadMovies()
+	}
+	//endregion
+
+	//region Public
+	fun requestMoreMovies() {
+		loading = true
+		currentPage++
+		loadMovies()
+	}
+
+	fun selectMovieAtIndex(index: Int) {
+		val movie = movieEntities[index]
+
+		if (selectedMovie.get() != null && movie.id == selectedMovie.get().id) {
+			selectedMovie.notifyChange()
+		} else {
+			val movieToShow = mapMovieToDetails(movie, imageConfiguration)
+
+			selectedMovie.set(movieToShow)
+		}
+	}
+	//endregion
+
+	//region Private
+	private fun loadMovies() {
+		if (genres.isEmpty()) {
```

Comment:
If `onStart` is called and there is no internet connection the genres list will be empty, so we need to make sure that there are genres and the image configuration before we fetch the movies.